### PR TITLE
`/auctions` APIs for crypto

### PR DESCRIPTION
**Pull request by correaswebert**
Creating the API for /auctions route under crypto.
You can create auctions, make bids, fetch auctions (by id and time)

API Contracts: https://github.com/Real-Dev-Squad/website-api-contracts/pull/37

Since this is a high priority feature, we'll have tests in other PR
Tests issue created and assigned #234

### CODE REVIEW

File: controllers/auctionController.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    logger.error(`Error fetching ongoing auctions: ${error}`)
+    return res.boom.badImplementation('An internal server error occured.')
+  }
+}
+
+const fetchAuctionById = async (req, res) => {
+  try {
+    const auctionId = req.params.id
+    const auctionData = await auctions.fetchAuctionById(auctionId)
+    logger.info(auctionData)
```
Comment (swarajpure) [inline controllers/auctionController.js:17]: Is line 17 needed?

File: controllers/auctionController.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    logger.error(`Error fetching ongoing auctions: ${error}`)
+    return res.boom.badImplementation('An internal server error occured.')
+  }
+}
+
+const fetchAuctionBySeller = async (req, res) => {
+  try {
+    const sellerId = req.params.id
+    const auctionsBySeller = await auctions.fetchAuctionBySeller(sellerId)
+    logger.info(auctionsBySeller)
```
Comment (swarajpure) [inline controllers/auctionController.js:29]: re: logger

File: controllers/auctionController.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  } catch (error) {
+    logger.error(`Error fetching ongoing auctions: ${error}`)
+    return res.boom.badImplementation('An internal server error occured.')
+  }
+}
+
+const createNewAuction = async (req, res) => {
+  try {
+    const { username: seller } = req.userData
+    const { initialPrice, item, duration, quantity } = req.body
```
Comment (swarajpure) [inline controllers/auctionController.js:40]: No validation on incoming req.body?

File: controllers/auctionController.js [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    return res.json(auctionsBySeller)
+  } catch (error) {
+    logger.error(`Error fetching ongoing auctions: ${error}`)
+    return res.boom.badImplementation('An internal server error occured.')
+  }
+}
+
+const createNewAuction = async (req, res) => {
+  try {
+    const { username: seller } = req.userData
```
Comment (swarajpure) [inline controllers/auctionController.js:39]: We could use `id` from the userData and save that it DB

File: controllers/auctionController.js [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+    return res.json({ id: auctionId, message: 'Auction created successfully!' })
+  } catch (error) {
+    logger.error(`Error creating new auctions: ${error}`)
+    return res.boom.badImplementation('An internal server error occured.')
+  }
+}
+
+const makeNewBid = async (req, res) => {
+  try {
+    const { username: bidderId } = req.userData
```
Comment (swarajpure) [inline controllers/auctionController.js:51]: re: `id`

File: controllers/auctionController.js [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+    const auctionId = req.params.id
+    const { bid } = req.body
```
Comment (swarajpure) [inline controllers/auctionController.js:53]: re: validation

File: controllers/auctionController.js [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+  } catch (error) {
+    logger.error(`Error fetching ongoing auctions: ${error}`)
+    return res.boom.badImplementation('An internal server error occured.')
+  }
+}
+
+const fetchAuctionBySeller = async (req, res) => {
+  try {
+    const sellerId = req.params.id
+    const auctionsBySeller = await auctions.fetchAuctionBySeller(sellerId)
```
Comment (swarajpure) [inline controllers/auctionController.js:28]: Need to be converted to `userId`

File: models/auctions.js [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+const makeNewBid = async ({ bidderId, auctionId, bid }) => {
+  try {
+    const bidRef = await bidsModel.add({
+      auctionId: auctionId,
+      bidderId: bidderId,
+      bid: bid,
+      time: new Date().getTime()
+    })
+
+    logger.info(`Made new bid ${bidRef.id}`)
```
Comment (swarajpure) [inline models/auctions.js:114]: re: logger

File: routes/auctions.js [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+
+router.get('/', auctionController.fetchAvailableAuctions)
+
+router.get('/seller/:id', auctionController.fetchAuctionBySeller)
+
+router.post('/:id', authenticate, auctionController.makeNewBid)
+router.post('/bid/:id', authenticate, auctionController.makeNewBid)
+
+router.post('/', authenticate, auctionController.createNewAuction)
+router.post('/create', authenticate, auctionController.createNewAuction)
```
Comment (swarajpure) [inline routes/auctions.js:16]: Let's not keep line number 12 and 16

Comment (swarajpure) [inline routes/auctions.js:16]: Can we have swagger docs?

**Review by swarajpure — COMMENTED** [review]
Great job! 🎉

**Review by ankurnarkhede — CHANGES_REQUESTED** [review]
@swarajpure , can you run the generate schema command, then we can merge this.

**Review by ankurnarkhede — APPROVED** [review]
Tests Issue: https://github.com/Real-Dev-Squad/website-backend/issues/236

**Comment by ankurnarkhede** [comment]
LGTM ✅

**Comment by ankushdharkar** [comment]
Implements Real-Dev-Squad/website-todo-items#16

**Pull request merged**